File: solutions/c/pangram/1/pangram.c
#include "pangram.h"

bool is_pangram(const char *sentence){
    if (sentence==NULL){
        return false;
    }
    int arr[26] = {0};
    for(size_t i=0;(unsigned char)sentence[i]!='\0';i++){
        if (isalpha((unsigned char)sentence[i])){
            char l = tolower((unsigned char)sentence[i]);
            arr[l-'a'] = 1;
        }
    }
    int j = 0;
    while (j<26){
        if (arr[j]<=0){
            return false;
        }
        j++;
    }
    return true;
}

File: solutions/go/pangram/1/pangram.go
package pangram

import(
    "strings"
    "unicode"
)

func IsPangram(input string) bool {
	input = strings.ToLower(input)
    a := make([]int,26)
    for _,x := range input{
        if unicode.IsLetter(x){
            a[x-'a']++
        }
    }
    for _,x := range a{
        if x==0{
            return false
        }
    }
    return true
}
